web/src/search.ts

```diff
@@ -151,17 +184,10 @@ export function initialize({on_narrow_search}: {on_narrow_search: OnNarrowSearch
                 return;
             }
 
-            if (keydown_util.is_enter_event(e) && $search_query_box.is(":focus")) {
-                // We just pressed Enter and the box had focus, which
-                // means we didn't use the typeahead at all.  In that
-                // case, we should act as though we're searching by
-                // terms.  (The reason the other actions don't call
-                // this codepath is that they first all blur the box to
-                // indicate that they've done what they need to do)
-
-                // Pill is already added during keydown event of input pills.
-                narrow_or_search_for_term(get_search_bar_text(), {on_narrow_search});
-                $search_query_box.trigger("blur");
```

Comment:
This already happens most of the time (when we want it to) in `narrow_or_search_for_term`